contribs/drt-extensions/src/main/java/org/matsim/contrib/drt/extension/edrt/scheduler/EmptyVehicleChargingScheduler.java

```diff
@@ -43,23 +45,25 @@
 public class EmptyVehicleChargingScheduler {
 	private final MobsimTimer timer;
 	private final EDrtTaskFactoryImpl taskFactory;
-	private final Map<Id<Link>, Charger> linkToChargerMap;
+	private final Map<Id<Link>, List<Charger>> linkToChargersMap;
 
 	public EmptyVehicleChargingScheduler(MobsimTimer timer, DrtTaskFactory taskFactory,
 			ChargingInfrastructure chargingInfrastructure) {
 		this.timer = timer;
 		this.taskFactory = (EDrtTaskFactoryImpl)taskFactory;
-		linkToChargerMap = chargingInfrastructure.getChargers()
+		linkToChargersMap = chargingInfrastructure.getChargers()
 				.values()
 				.stream()
-				.collect(Collectors.toMap(c -> c.getLink().getId(), c -> c));
+				.collect(Collectors.groupingBy(c -> c.getLink().getId()));
 	}
 
 	public void chargeVehicle(DvrpVehicle vehicle) {
 		DrtStayTask currentTask = (DrtStayTask)vehicle.getSchedule().getCurrentTask();
 		Link currentLink = currentTask.getLink();
-		Charger charger = linkToChargerMap.get(currentLink.getId());
-		if (charger != null) {
+		List<Charger> chargers = linkToChargersMap.get(currentLink.getId());
+		if (chargers != null) {
+			// Take the charger with the smallest queue
+			Charger charger = chargers.stream().min(Comparator.comparing(e -> e.getLogic().getQueuedVehicles().size())).orElseThrow();
```

Comment:
Replace `comparing()` with `comparingInt()`

Reply:
You can also change
```
min(Comparator.comparing(e -> e.getLogic().getQueuedVehicles().size()))
```
to something like this:
```
mapToInt(e -> e.getLogic().getQueuedVehicles().size()).min()
```

Reply:
`mapToInt(e -> e.getLogic().getQueuedVehicles().size()).min()` but we want to get the Charger with the lowest number of queued vehicles. To object of intreset is charger not the int.

Reply:
True:-) Suggesting code changes without trying them in an IDE...